dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java

```diff
@@ -2565,7 +2563,7 @@ public int saveProcessDefine(User operator, ProcessDefinition processDefinition,
     @Override
     public int saveTaskRelation(User operator, long projectCode, long processDefinitionCode, int processDefinitionVersion,
                                 List<ProcessTaskRelationLog> taskRelationList, List<TaskDefinitionLog> taskDefinitionLogs,
-                                Boolean syncDefine) {
+                                Boolean syncDefine, boolean isDelete) {
```

Comment:
We should avoid this way, because `saveTaskRelation` is already a complex function, expanding the `isDelete` model will exponentially increase the maintenance cost.

Reply:
I think it will defaultly delete the old relation list and insert the new ones. 
What's the case will it not be deleted?

Reply:
When adding a task definition, the association of all workflow task definitions will be cleared. This is wrong